### PR TITLE
[HCPVault] Docs only integration tile

### PR DESCRIPTION
**Pull request by ls339**
### What does this PR do?

Create an integration tile for HCPVault

### Motivation

New integration tile and dashboard for metrics emitted from HCPVault

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: hcp_vault/assets/dashboards/hcp_vault_overview.json [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+                "type": "group",
+                "banner_img": "/static/images/logos/vault_large.svg",
+                "show_title": false,
+                "layout_type": "ordered",
+                "widgets": [
+                    {
+                        "id": 5633470244696532,
+                        "definition": {
+                            "type": "note",
+                            "content": "[HCP Vault Cluster Link](https://portal.cloud.hashicorp.com/services/vault/$cluster.value?project_id=$project.value)\n\nWith HCP Vault, you can leverage a SaaS service with secret management and encryption capabilities. This dashboard provides a high-level overview of your Vault clusters so you can monitor its performance and cluster health.",
```
Comment (kayayarai) [inline hcp_vault/assets/dashboards/hcp_vault_overview.json:19]:
```suggestion
                            "content": "[HCP Vault Cluster Link](https://portal.cloud.hashicorp.com/services/vault/$cluster.value?project_id=$project.value)\n\nWith HCP Vault, you can leverage a SaaS service with secret management and encryption capabilities. This dashboard provides a high-level overview of your Vault clusters so you can monitor performance and cluster health.",
```

File: hcp_vault/assets/dashboards/hcp_vault_overview.json [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+                            "tick_edge": "left",
+                            "has_padding": true
+                        },
+                        "layout": {"x": 0, "y": 0, "width": 5, "height": 2}
+                    },
+                    {
+                        "id": 8127281118370152,
+                        "definition": {
+                            "type": "note",
+                            "content": "- [Official Vault integration docs](https://docs.datadoghq.com/integrations/vault/)\n- [Monitoring HashiCorp Vault with Datadog](https://www.datadoghq.com/blog/monitor-hashicorp-vault-with-datadog/)",
```
Comment (kayayarai) [inline hcp_vault/assets/dashboards/hcp_vault_overview.json:35]:
```suggestion
                            "content": "- [Vault integration documentation](https://docs.datadoghq.com/integrations/vault/)\n- [Monitoring HashiCorp Vault with Datadog](https://www.datadoghq.com/blog/monitor-hashicorp-vault-with-datadog/)",
```

**Review by kayayarai — CHANGES_REQUESTED** [review]

File: hcp_vault/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# HCP Vault Integration
+
+## Overview
+
+The HCP Vault integration provides an overview of your Vault clusters so you can monitor its performance and cluster health.
+
+Availability: HCP Vault metrics streaming is available for all production grade cluster tiers.  The feature is not available for Development tier clusters.
+
+For details on metrics scope and interpretation, see the [HCP Vault Metrics Guidance][1]
```
[the remark this answers is not in the document]
Comment (yzhan289) [reply hcp_vault/README.md:9]: Fixed link

**Review by yzhan289 — APPROVED** [review]
LGTM

**Review by kayayarai — APPROVED** [review]

**Pull request merged**